### PR TITLE
bootstrap: unify tenant and sys tables

### PR DESCRIPTION
**Pull request by dt**
We want to be able to replicate a system tenant into a non-system tenant
but this requires that they have the same set of system tables so that
when the non-system tenant starts executing, it finds all the same tables
that it expects to exist, regardless of whether it was created via PCR of
a system tenant or bootstrap of a secondary tenant.

Release note: none.
Epic: none.

### CODE REVIEW

File: pkg/sql/catalog/bootstrap/metadata.go [hunk 1]
```diff
@@ -101,18 +101,12 @@ func (ms *MetadataSchema) AddDescriptor(desc catalog.Descriptor) {
 // AddDescriptorForSystemTenant is like AddDescriptor but only for the system
 // tenant.
 func (ms *MetadataSchema) AddDescriptorForSystemTenant(desc catalog.Descriptor) {
-	if !ms.codec.ForSystemTenant() {
-		return
-	}
 	ms.AddDescriptor(desc)
 }
```
Comment (stevendanna) [inline pkg/sql/catalog/bootstrap/metadata.go]: We may want to remove these functions and use AddDescriptor directly. With this change the comments are lying a bit.

**Review by stevendanna — COMMENTED** [review]
Some commentary in the commit message about why we think this is reasonable might be useful. I imagine there is a long-list of changes needed to make the tests pass so I'll give this another look when those have settled a bit.

**Comment by dt** [comment]
oh yeah, this was just to let CI tell me what is broken. I plan to add the migration to this (and commentary) before it is ready to go.

**Comment by dt** [comment]
> Some commentary in the commit message about why we think this is reasonable might be useful.

Done.

> a long-list of changes needed to make the tests pass

Done.

**Review by stevendanna — APPROVED** [review]

**Comment by dt** [comment]
TFTR

bors r+

**Comment by dt** [comment]
blathers backport 24.1

**Pull request merged**